### ISSUE TITLE
WebAPI Usage

### ISSUE THREAD

**Issue by ddahya** [simonhebert/nx595e-web-api#1]
Hey,  this is not an issue but rather a request on how to use your webapi.  I also have the NX595e module in my alarm. 

Are you able to provide some instruction on how the webapi connects back to NX595 module.
Thanks.

**Comment by simonhebert** [comment]
This is a new code project where I will send the source code for the API that is already running in my network in still test mode for now. Stay tuned, I will update it soon. The API connect to the NX-595E module by the Web interface on it that I have reverse engineered to communicate with.

**Comment by ddahya** [comment]
Hi Simon,

Just wanted to know how you were going with your code?  do you have any time frames on sharing your project... I'm wanting to integrate my NX595 into my home automation.

Thanks.

**Comment by simonhebert** [comment]
Hey ddahya! Look at this, I finally deployed the source code of my last API version running on my network since a few months. This could help you. Sorry for the delay, I ran out of time for that.

**Comment by ddahya** [comment]
Hi Simon,

Thanks for the updated code. I’ve been trying to get your web-api to connect to my NX595e interface without any success.

I’ve tried on both Windows and Linux machines but unable to get it to work. The only change I made was in the appsettings (Host, LoginName and Password) to reflect my own settings.

I have run the code in VS2017 in debug mode and I’m getting the following messages, details are in the attached txt file.

[NX595 Debug.txt](https://github.com/simonhebert/nx595e-webapi/files/2903813/NX595.Debug.txt)

Any chance you could share your container config, that what ill like to do once I get this working. 

Really appreciate all your hard work.

**Comment by simonhebert** [comment]
The error seem to be related to the status result returned by your NX-595E interface that maybe is different of mine. How many partition do you have in your system and how many zones? I have only one partition and 8 zones used. The system can't parse and convert the string in integer.

**Comment by ddahya** [comment]
Thanks for the quick reply,  In my NX595E Interface i have one Partition and 16 Zones configured.

**Comment by simonhebert** [comment]
Could you connect to the NX-595E web interface, go to the "Zones" section, and send me the HTML source code returned by the zones web page? I will try to figure the difference between my 8 zones display and your 16 zones display.

**Comment by ddahya** [comment]
Here is the source code from my 16 Zones on the NX595E

[zone.txt](https://github.com/simonhebert/nx595e-webapi/files/2908608/zone.txt)

**Comment by simonhebert** [comment]
There is a difference between the code produced by your interface (named ComNav?) and the standard NX-595E interface I own if I compare the sources:

![image](https://user-images.githubusercontent.com/44216662/53534599-896f7300-3acd-11e9-9502-9017c8e06e2b.png)

Since this code work by reverse engineering the system, the controller is only able to understand the standard NX-595E for now. Can you tell me the details of your module card in your alarm system?

**Comment by ddahya** [comment]
I have a Hill ComNav module, sold in AUS/NZ (just spoke to the place i brought it from) its a re branded UTC/Caddx NX-595E card.  

Here are some details of the card. 

![image](https://user-images.githubusercontent.com/20119625/53538067-10295c00-3b71-11e9-9d41-8e925bdb935c.png)

The module is connected to a Hills Reliance R12

**Comment by simonhebert** [comment]
Yes, it's what I've found about this module card in my searchs to. I have the documentation about it. There is some custom software personnalisation for this brand and the web interface version seem older than the NX-595E I own. (ComNav v_CN_0.106-j vs NX-595E v_CN_0.108-O)

This is the cause of the error produced by the Web API. The difference between the interfaces make the controller unable to parse some data. Also, no evidence that the command sent by the Web API to the interface will be understand by your card like to arm or disarm.

We will need to do some test to validate. I will try to modify the code to parse the data differently depending on the brand/version of the module used to see if I can resolve the issue.

**Comment by ddahya** [comment]
Just let me know what you require from me re testing..

**Comment by ddahya** [comment]
Hey Simon,  Just wanted to know if you have had any chance at looking at updating your code for my ComNav?

Thanks
